docs/content/guides/dagster-pipes/kubernetes.mdx

```diff
@@ -0,0 +1,243 @@
+---
+title: "Integrating Kubernetes with Dagster Pipes | Dagster Docs"
+description: "Learn to integrate Dagster Pipes with Kubernetes to launch external code from Dagster assets."
+---
+
+# Integrating Kubernetes with Dagster Pipes
+
+<Note>
+  <strong>Heads up!</strong> This guide focuses on using an out-of-the-box
+  Kubernetes resource. For further customization, use the{" "}
+  <a href="/guides/dagster-pipes/customizing-dagster-pipes-protocols">
+    <code>open_pipes_session</code> approach
+  </a>{" "}
+  instead.
+</Note>
+
+In this guide, we’ll show you how to use [Dagster Pipes](/guides/dagster-pipes) with Dagster’s Kubernetes integration to launch Kubernetes pods and execute external code.
+
+Pipes allows your Kubernetes code to execute outside of a full Dagster environment. Instead, the environment only needs to contain `dagster-pipes`, a single-file Python package with no dependencies that can be installed from PyPI or easily vendored. `dagster-pipes` handles streaming `stdout`/`stderr` and Dagster events back to the orchestration process.
+
+---
+
+## Prerequisites
+
+To use Dagster Pipes with Kubernetes, you’ll need:
+
+- **In the orchestration environment**, you'll need to install the following packages:
+
+  ```shell
+  pip install dagster dagster-webserver dagster-k8s kubernetes
+  ```
+
+  Refer to the [Dagster installation guide](/getting-started/install) for more info.
+
+- **In Kubernetes:**
+  - **An existing Kubernetes cluster.**
+  - **The following information about your Kubernetes cluster**:
+    - `config_file` - The name of the Kubernetes config file
+    - `context` - The name of the Kubernetes context you want to use
+    - `image` - The image you want to run using Dagster Pipes
+
+---
+
+## Step 1: Create an asset computed in Kubernetes
+
+In this step, you’ll create a Dagster asset that, when materialized, opens a Dagster pipes session and spins up a Kubernetes pod that executes some external code.
+
+### Step 1.1: Define the Dagster asset
+
+In your Dagster project, create a file named `dagster_k8s_pipes.py` and paste in the following code:
+
+```python
+# dagster_k8s_pipes.py
+
+from dagster import AssetExecutionContext, Definitions, asset
+from dagster_k8s import PipesK8sClient
+
+
+@asset
+def k8s_pipes_asset(context: AssetExecutionContext, k8s_pipes_client: PipesK8sClient):
+  return k8s_pipes_client.run(
```

Comment:
Updated this to use the new config options mentioned in last review. 

One thing I'm still not clear on is how does Pipes know which code/file to run - how do users specify this? Is it done here in the asset, or in Kubernetes itself?

Reply:
ya I think we can drop `kubeconfig_file` `kube_context`

> One thing I'm still not clear on is how does Pipes know which code/file to run - how do users specify this? Is it done here in the asset, or in Kubernetes itself?

this is the `image` argument [1], which is related to [2]

Reply:
That's what I thought, but it seemed like there was something else to do in k8s so I wasn't sure. We'll chat through this later!